python/ray/train/xgboost/config.py

```diff
@@ -0,0 +1,99 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from xgboost import RabitTracker
+
+import ray
+from ray.train._internal.worker_group import WorkerGroup
+from ray.train.backend import Backend, BackendConfig
+
+
+@dataclass
+class XGBoostConfig(BackendConfig):
+    """Configuration for xgboost collective communication setup.
+
+    Args:
+        xgboost_communicator: The backend to use for collective communication for
+            distributed xgboost training. One of ["rabit", "federated"].
+        args: Additional arguments to pass to the backend.
+            See ___. TODO: link to __init__ docstring
+    """
+
+    xgboost_communicator: Optional[str] = "rabit"
+    args: Optional[Dict[str, Any]] = None
+
+    @property
+    def backend_cls(self):
+        return _XGBoostBackend
+
+
+class _XGBoostBackend(Backend):
+    def on_start(self, worker_group: WorkerGroup, backend_config: XGBoostConfig):
+        user_args = backend_config.args or {}
+        xgboost_communicator = user_args.pop(
+            "xgboost_communicator", backend_config.xgboost_communicator
+        )
+        if xgboost_communicator == "rabit":
+            # Set up the rabit tracker on the Train driver.
+            num_workers = len(worker_group)
+            self.rabit_args = {"DMLC_NUM_WORKER": num_workers}
```

Comment:
nit: Move all instance variables into an `__init__` method so they're easier to keep track of?